hyperpay/views.py

```diff
@@ -248,21 +269,19 @@ def get(self, request, encrypted_resource_path=None):
         check_status = self._get_check_status(request)
 
         try:
-            status = PaymentStatus.PENDING
             if check_status:
                 verification_response, status = self._verify_status(resource_path)
-                if (verification_response and isinstance(verification_response, dict) and
-                        verification_response.get('merchantTransactionId')):
-                    transaction_id = verification_response['merchantTransactionId']
+
+            if verification_response and 'merchantMemo' in verification_response:
+                transaction_id = verification_response['merchantMemo']
+                basket_id = OrderNumberGenerator().basket_id(transaction_id)
+                basket = self._get_basket(basket_id)
 
             if status == PaymentStatus.FAILURE:
                 return redirect(reverse('payment_error'))
-            if status == PaymentStatus.PENDING:
+            elif status == PaymentStatus.PENDING:
                 return self._handle_pending_status(request, encrypted_resource_path, resource_path)
 
-            basket_id = OrderNumberGenerator().basket_id(verification_response['merchantMemo'])
-            basket = self._get_basket(basket_id)
-
             transaction_id = verification_response['id']
```

Comment:
I don't understand why not all transaction_id are with the id?  I mean this line is not inside an else, and it's the last one that changes.
This also happens in the current main implementation.
The only reason is that something else failed before this line and transaction_id is set to `verification_response['merchantxxxx']`

Reply:
Actully the correct basket submitted are set with the id that send processor.
![2023-09-07_13-41](https://github.com/nelc/ecommerce-hyperpay/assets/51926076/1dddcd8c-2bf3-4e0c-99bc-d0bd1dd3927d)
And the failed to save the `merchantxxx` in the transaction_id.
![image](https://github.com/nelc/ecommerce-hyperpay/assets/51926076/0ca46aa8-4321-4050-af9d-41f36faf9368)

Reply:
me neither

Reply:
@andrey-canon what do you think in that behavior is ok that good transaction keep the id and the others keep the 'merchantxxxx' ??

Reply:
personally I prefer to keep the same transaction id for all the operations, it's easier to filter error and success cases, however that is out of the scope of this pr and I would like to keep changes at minimum